Library/Homebrew/formula.rb

```diff
@@ -2380,6 +2380,31 @@ def self.[](name)
     Formulary.factory(name)
   end
 
+  sig { params(name: String).returns(Formula) }
+  def self.stub(name)
+    Formulary.factory(name, prefer_stub: true)
+  end
+
+  sig { params(name: String).returns(T::Boolean) }
+  def self.any_version_installed?(name)
+    stub(name).any_version_installed?
+  rescue FormulaUnavailableError
+    false
+  end
+
+  sig {
+    params(
+      name:             String,
+      reason:           String,
+      latest:           T::Boolean,
+      output_to_stderr: T::Boolean,
+      quiet:            T::Boolean,
+    ).void
+  }
+  def self.ensure_installed!(name, reason: "", latest: false, output_to_stderr: true, quiet: false)
```

Comment:
Do we need this, or can we just adjust callers to use `Formula.stub("...").ensure_installed!` instead of `Formula["..."].ensure_installed!`?